yabt/buildcontext.py

```diff
@@ -456,19 +456,38 @@ def build(target: Target):
                 built_targets.add(target.name)
                 target.done()
 
-                # TODO: retry flaky tests N times
+                # TODO: retry flaky tests N times AFTER all the rest passed
                 # TODO: collect stats and print report at the end
                 # TODO: support both "fail fast" (exit on first failure) and
                 #       run all tests (don't exit on first failure) modes
                 target_tested = False
                 if run_tests and 'testable' in target.tags:
                     if self.conf.no_test_cache or not test_cached:
                         logger.info('Testing target {}', target.name)
-                        test_start = time()
-                        self.test_target(target)
+                        retries = (
+                            target.props.retries or self.conf.test_retries)
+                        if retries < 0:
+                            raise ValueError(
+                                'Retries value must be positive: got {}'
+                                .format(retries))
+                        test_start = 0
```

Comment:
why the useless init here?

Reply:
I dunno python...
Also, I assign value in the loop, and in cases where we don't reach the loop it'll error (of course if the loop won't run once we already thrown an exception).
I can remove it but I don't like unassigned values :(

Reply:
as you wish